cmdpolling/polling.go

```diff
@@ -1,15 +1,153 @@
 package cmdpolling
 
 import (
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
 	"github.com/codegangsta/cli"
+	"github.com/ingrammicro/concerto/api/polling"
+	"github.com/ingrammicro/concerto/cmd"
+	"github.com/ingrammicro/concerto/utils"
+	"github.com/ingrammicro/concerto/utils/format"
+)
+
+const (
+	PollingPingTimingInterval = 2
+	ProcessIdFile             = "imco-polling.pid"
+)
+
+var (
+	wg               = &sync.WaitGroup{}
+	commandProcessed = make(chan bool, 1)
 )
 
-func SubCommands() []cli.Command {
-	return []cli.Command{
-		{
-			Name:   "register",
-			Usage:  "Register concerto agent within an imported Host",
-			Action: cmdRegister,
-		},
+// Handle signals
+func handleSysSignals() {
+	log.Debug("handleSysSignals")
+
+	gracefulStop := make(chan os.Signal, 1)
+	signal.Notify(gracefulStop, syscall.SIGTERM, syscall.SIGINT, syscall.SIGKILL)
+	log.Debug("Ending, signal detected:", <-gracefulStop)
+	wg.Done()
+}
+
+// Start the polling process
+func cmdStart(c *cli.Context) error {
+	log.Debug("cmdStart")
+
+	formatter := format.GetFormatter()
+	if err := utils.SetProcessIdToFile(ProcessIdFile); err != nil {
+		formatter.PrintFatal("cannot create the pid file", err)
+	}
+
+	go handleSysSignals()
+
+	wg.Add(1)
+	go pingRoutine(c)
+	wg.Wait()
```

Comment:
Since you are going to be having the main goroutine blocked by the wait, why not do the following?
- run the pingRoutine func in the main goroutine
- have the handleSysSignals accept a context.CancelFunc it will call when a signal is received
- have the pingRoutine (and the goroutines it launches) accept a context.Context and check in every iteration whether  something can be received from the ctx.Done() channel/ ctx.Err() returns non nil?

That way you could avoid the global wg you are using now....